presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,529 @@
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Connector access control which uses existing Ranger policies for authorizations
+ */
+
+public class RangerBasedAccessControl
+        implements ConnectorAccessControl
+{
+    private static final Logger log = Logger.get(RangerBasedAccessControl.class);
+
+    private RangerAuthorizer rangerAuthorizer;
+    private Users users;
+    private Map<String, Set<String>> userRolesMapping = new HashMap<>();
+    private Map<String, Set<String>> userGroupsMapping = new HashMap<>();
+
+    public RangerBasedAccessControl(RangerBasedAccessControlConfig config)
+    {
+        requireNonNull(config, "config is null");
+        requireNonNull(config.getRangerHttpEndPoint(), "Ranger service http end point is null");
+        requireNonNull(config.getRangerHiveServiceName(), "Ranger hive service name is null");
+
+        ServicePolicies servicePolicies;
+        try {
+            OkHttpClient client = getAuthHttpClient(config);
+            HttpUrl hiveServicePolicyUrl = requireNonNull(HttpUrl.get(URI.create(config.getRangerHttpEndPoint())))
+                    .newBuilder()
+                    .encodedPath(RANGER_REST_POLICY_MGR_DOWNLOAD_URL + "/" + config.getRangerHiveServiceName()).build();
+
+            HttpUrl getUsersUrl = requireNonNull(HttpUrl.get(URI.create(config.getRangerHttpEndPoint())))
+                    .newBuilder()
+                    .encodedPath(RANGER_REST_USER_GROUP_URL)
+                    .build();
+
+            servicePolicies = getHiveServicePolicies(client, hiveServicePolicyUrl);
+            users = getUsers(client, getUsersUrl);
+            userGroupsMapping = getuserGroupsMappings(users);
+            rangerAuthorizer = new RangerAuthorizer(servicePolicies, config);
+            userRolesMapping = getRolesForUserList(client, config.getRangerHttpEndPoint());
+        }
+        catch (Exception e) {
+            log.error(e, "Exception while querying ranger service ");
+            throw new AccessDeniedException("Exception while querying ranger service ");
+        }
+    }
+
+    @VisibleForTesting
+    public RangerBasedAccessControl(RangerAuthorizer rangerAuthorizer, Map<String, Set<String>> userGroupsMapping, Map<String, Set<String>> userRolesMapping)
+    {
+        this.rangerAuthorizer = rangerAuthorizer;
+        this.userGroupsMapping = userGroupsMapping;
+        this.userRolesMapping = userRolesMapping;
+    }
+
+    private ServicePolicies getHiveServicePolicies(OkHttpClient client, HttpUrl httpUrl) throws JsonProcessingException
+    {
+        Request request = new Request.Builder().url(httpUrl).header("Accept", "application/json").build();
+
+        JsonCodec<ServicePolicies> policiesJsonCodec = jsonCodec(ServicePolicies.class);
+        JsonResponse<ServicePolicies> servicePolicies = JsonResponse.execute(policiesJsonCodec, client, request);
```

Comment:
As I tested - `JsonResponse` is unable to work with `ServicePolicies` and hence we need to use Object Mapper, so in this situation it is not useful to use either of `JsonCodec` or `JsonResponse` here